src/Validator/ValidationNode.php

```diff
@@ -0,0 +1,152 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Overblog\GraphQLBundle\Validator;
+
+use GraphQL\Type\Definition\InputObjectType;
+use GraphQL\Type\Definition\ObjectType;
+use GraphQL\Type\Definition\ResolveInfo;
+use GraphQL\Type\Definition\Type;
+use Overblog\GraphQLBundle\Definition\Argument;
+
+/**
+ * ValidationNode.
+ *
+ * Holds the input data of the associated to it GraphQL type. Properties will be
+ * created dinamically in runtime. In order to avoid name conflicts all built in
+ * property names are prefixed with double underscores.
+ *
+ * It also contains variables of the resolver context, in which this class was
+ * instantiated.
+ */
+class ValidationNode
+{
+    private const KNOWN_VAR_NAMES = ['value', 'args', 'context', 'info'];
+
+    /**
+     * @var ObjectType|InputObjectType|Type
+     */
+    private $__type;
```

Comment:
why the prefix `__` for vars name?

Reply:
The objects of the class ValidationNode are populated with user input data to be validated later. The properties are created dynamically from argument names of GraphQL types. The double underscore prefix should prevent name conflicts, if a user uses in his types such names as ‘type’ or ‘fieldName’ etc.